src/firefly/js/visualize/region/RegionFactory.js

```diff
@@ -0,0 +1,628 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import { RegionType, RegionCsys, RegionValueUnit,
+         RegionOptions, RegionValue, RegionDimension, RegionPointType, RegParseException,
+         getRegionType, getRegionCoordSys, getRegionPointType, makeFont,
+         makePoint, makeText, makeBox, makeBoxAnnulus, makeAnnulus, makeCircle, makeEllipse, makeEllipseAnnulus,
+         makeLine, makePolygon} from './Region.js';
+import validator from 'validator';
+import {CoordinateSys} from '../CoordSys.js';
+import {makeWorldPt} from '../Point.js';
+import {convertAngle} from '../VisUtil.js';
+
+var RegionParseError = {
+    ErrSyntax: 'wrong region syntax',
+    InvalidCoord: 'region coordinate undefined',
+    InvalidType:  'region type undefined',
+    InvalidParam:  'invalid region syntax',
+    InvalidProp: 'invalid region properties'
+};
+
+export class RegionFactory {
+
+    static parseRegionJson(RegionData) {
+        try {
+            return RegionData.reduce ( (prev, region, index) => {
+                const rg = RegionFactory.parsePart(region, index);  // skip comment line
+                if (rg) {
+                    prev.push(rg);
+                }
+                return prev;
+            }, []);
+        }
+        catch (e) {
+            throw e;
+        }
+    }
+
+    /**
+     * parsePart parses the region data of JSON result (one item from RegionData array)
+     * @param regionStr
+     * @param index
+     * @returns {Region} or throw RegParseException
+     */
+    static parsePart(regionStr, index = -1) {
+
```

Comment:
I am not convinced we need to throw an exception anymore, since we are only supporting parsePart.  You could just `return false`, return an object with a RegionType of some sort of error and a message key. I get more nervous about throwing exceptions in JavaScript since catching is not enforced.